cpu/cortexm_common/Makefile.dep

```diff
@@ -0,0 +1,19 @@
+# Tell the build system that the CPU depends on the Cortex-M common files:
+USEMODULE += cortexm_common
+
+# include common periph code
+USEMODULE += cortexm_common_periph
+
+# all cortex MCU's use newlib as libc
+USEMODULE += newlib
+
+# use the nano-specs of Newlib when available
+USEMODULE += newlib_nano
+
+# Export the peripheral drivers to be linked into the final binary:
+USEMODULE += periph
+
+# Use Hardware FPU by default if present
+ifneq (,$(filter cortexm_fpu,$(FEATURES_PROVIDED)))
+  DEFAULT_MODULE += cortexm_fpu
+endif
```

Comment:
What about `FEATURES_OPTIONAL += cortexm_fpu`, it will get used when provided. It could be added to DEFAULT_MODULE depnding on that check, so no to check `FEATURES_PROVIDED` but `FEATURES_USED`

Reply:
Changed, now it checks for `FEATURES_USED`.